Assets/GameAssets/Player/PlayerScripts/Harpoon/HarpoonGun.cs

```diff
@@ -275,14 +294,40 @@ private void HarpoonFullyReloaded()
     {
         _harpoonFiringState = EHarpoonFiringState.Ready;
         _harpoonOnGun.SetActive(true);
+        RuntimeSfxManager.APlayOneShotSfx?
+            .Invoke(FmodSfxEvents.Instance.HarpoonReload, gameObject.transform.position);
 
-        if(_isFocusButtonHeld)
+        if (_isFocusButtonHeld)
         {
             StartFocusingHarpoon();
         }
 
         PlayerManager.Instance.InvokeOnHarpoonReloadedEvent();
     }
+
+    /// <summary>
+    /// Restocks harpoons to full from the given ammo rack
+    /// </summary>
+    /// <param name="ammoRack"> The ammo rack to take harpoons from </param>
+    private void RestockHarpoons(AmmoRackInteractable ammoRack)
+    {
+        int targetAmmo = 0;
+
+        if (_harpoonFiringState != EHarpoonFiringState.Ready)
+        {
+            targetAmmo = _maxAmmo;
+        }
+        else
+        {
+            targetAmmo = _maxAmmo - 1;
+        }
+
+        while (!ammoRack.OutOfHarpoons() && _currentReserveAmmo < targetAmmo)
```

Comment:
More of a preference, but I feel this could be done by just checking the lesser of how many more harpoons can be picked up and how many more harpoons remain and just adding that many. Not a required change, but that makes more sense to me

Reply:
If it's all the same to you, I'd like to keep this as is. I'm sitting here trying to come up with another solution, but I think I like this one the best. Although it is taking the harpoons "one by one," I can't think of another way to solve this problem without using a loop similar to this along the way.